tests/foreman/maintain/test_advanced.py

```diff
@@ -22,67 +22,26 @@
 from robottelo.config import robottelo_tmp_dir
 from robottelo.config import settings
 from robottelo.constants import MAINTAIN_HAMMER_YML
+from robottelo.constants import SAT_NON_GA_VERSIONS
+from robottelo.hosts import get_sat_rhel_version
+from robottelo.hosts import get_sat_version
 
-pytestmark = pytest.mark.destructive
-
-
-# Common repositories for Satellite and Capsule
-common_repos = ['rhel-8-for-x86_64-baseos-rpms', 'rhel-8-for-x86_64-appstream-rpms']
-
-# Satellite repositories
-sat_611_repos = [
-    'satellite-6.11-for-rhel-8-x86_64-rpms',
-    'satellite-maintenance-6.11-for-rhel-8-x86_64-rpms',
-] + common_repos
-
-sat_612_repos = [
-    'satellite-6.12-for-rhel-8-x86_64-rpms',
-    'satellite-maintenance-6.12-for-rhel-8-x86_64-rpms',
-] + common_repos
-
-sat_613_repos = [
-    'satellite-6.13-for-rhel-8-x86_64-rpms',
-    'satellite-maintenance-6.13-for-rhel-8-x86_64-rpms',
-] + common_repos
-
-sat_614_repos = [
-    'satellite-6.14-for-rhel-8-x86_64-rpms',
-    'satellite-maintenance-6.14-for-rhel-8-x86_64-rpms',
-] + common_repos
-
-# Capsule repositories
-cap_611_repos = [
-    'satellite-capsule-6.11-for-rhel-8-x86_64-rpms',
-    'satellite-maintenance-6.11-for-rhel-8-x86_64-rpms',
-] + common_repos
-
-cap_612_repos = [
-    'satellite-capsule-6.12-for-rhel-8-x86_64-rpms',
-    'satellite-maintenance-6.12-for-rhel-8-x86_64-rpms',
-] + common_repos
-
-cap_613_repos = [
-    'satellite-capsule-6.13-for-rhel-8-x86_64-rpms',
-    'satellite-maintenance-6.13-for-rhel-8-x86_64-rpms',
-] + common_repos
-
-cap_614_repos = [
-    'satellite-capsule-6.14-for-rhel-8-x86_64-rpms',
-    'satellite-maintenance-6.14-for-rhel-8-x86_64-rpms',
-] + common_repos
-
-sat_repos = {
-    '6.11': sat_611_repos,
-    '6.12': sat_612_repos,
-    '6.13': sat_613_repos,
-    '6.14': sat_614_repos,
-}
-cap_repos = {
-    '6.11': cap_611_repos,
-    '6.12': cap_612_repos,
-    '6.13': cap_613_repos,
-    '6.14': cap_614_repos,
-}
+
+sat_x_y_release = f'{get_sat_version().major}.{get_sat_version().minor}'
+
+
+def get_satellite_capsule_repos(
```

Comment:
Do you see this could be moved to / mixed with (atleast for RHEL repos) the https://github.com/SatelliteQE/robottelo/blob/master/robottelo/host_helpers/contenthost_mixins.py#L14 repo related cached properties! 

Not forcing you to do it in this PR though !

Reply:
#12492 created to track this request